detect_secrets/plugins/private_key.py

```diff
@@ -58,6 +64,39 @@ class PrivateKeyDetector(RegexBasedDetector):
         re.compile(r'PuTTY-User-Key-File-2'),
     ]
 
+    def __init__(self) -> None:
+        self._analyzed_files: Set[str] = set()
+
+    def analyze_line(
+            self,
+            filename: str,
+            line: str,
+            line_number: int = 0,
+            context: Optional[CodeSnippet] = None,
+            raw_context: Optional[CodeSnippet] = None,
+            **kwargs: Any
+    ) -> Set[PotentialSecret]:
+        output: Set[PotentialSecret] = set()
+
+        output.update(
```

Comment:
why do you need this part if you already read the entire file in the next part?

Reply:
In some cases for private keys the line already contains a violation, and we don't need the file reading

Reply:
It is similar to how our multiline detectors work today